spec/mem/tree_borrows/memory.md

```diff
@@ -65,15 +65,31 @@ impl TreeBorrowsFrameExtra {
 Here we define some helper methods to implement the memory interface.
 
 ```rust
+pub struct NewPermission {
+    /// Permission for the frozen part of the range.
+    freeze_perm: Permission,
+    /// Whether a read access should be performed on the frozen part on a retag.
+    freeze_access: bool,
+    /// Permission for the non-frozen part of the range.
+    nonfreeze_perm: Permission,
+    /// Whether a read access should be performed on the non-frozen
+    /// part on a retag.
+    nonfreeze_access: bool,
+    /// Whether this pointer is part of the arguments of a function call.
+    /// `protector` is `Some(_)` for all pointers marked `noalias`.
+    protected: Protected,
+}
+
+
 impl<T: Target> TreeBorrowsMemory<T> {
     /// Create a new node for a pointer (reborrow)
     fn reborrow(
-        &mut self, 
+        &mut self,
         ptr: ThinPointer<TreeBorrowsProvenance>,
```

Comment:
to properly support unsized types, the `ptr` argument here has to become a `Pointer` instead of a `ThinPointer`, as it needs the `PointerMeta`. We can do this refactor later.